DigitalLearningSolutions.Web.AutomatedUiTests/AccessibilityTests.cs

```diff
@@ -42,13 +42,55 @@ public void Page_has_no_accessibility_errors(string url)
             axeResult.Violations.Should().BeEmpty();
         }
 
+        // TODO: enable this test when AutomatedUiTests can connect to the test database
+        // [Fact]
+        public void Registration_journey_has_no_accessibility_errors()
+        {
+            // given
+            var registerUrl = "/Register";
+
+            // when
+            driver.Navigate().GoToUrl(baseUrl + registerUrl);
+            var registerResult = new AxeBuilder(driver).Analyze();
+            var firstName = driver.FindElement(By.Id("FirstName"));
+            firstName.SendKeys("Test");
+            var lastName = driver.FindElement(By.Id("LastName"));
+            lastName.SendKeys("User");
+            var email = driver.FindElement(By.Id("Email"));
+            email.SendKeys("email@test.com");
+            var registerForm = driver.FindElement(By.TagName("form"));
+            registerForm.Submit();
+
+            var learnerInformationResult = new AxeBuilder(driver).Analyze();
+            var centre = new SelectElement(driver.FindElement(By.Id("Centre")));
+            centre.SelectByValue("2");
+            var jobGroup = new SelectElement(driver.FindElement(By.Id("JobGroup")));
+            jobGroup.SelectByValue("1");
+            var learnerInformationForm = driver.FindElement(By.TagName("form"));
+            learnerInformationForm.Submit();
+
+            var passwordResult = new AxeBuilder(driver).Analyze();
+            var password = driver.FindElement(By.Id("Password"));
+            password.SendKeys("password!1");
+            var confirmPassword = driver.FindElement(By.Id("ConfirmPassword"));
+            confirmPassword.SendKeys("password!1");
+            var passwordForm = driver.FindElement(By.TagName("form"));
+            passwordForm.Submit();
+
+            var summaryResult = new AxeBuilder(driver).Analyze();
+
+            // then
+            registerResult.Violations.Should().BeEmpty();
+            learnerInformationResult.Violations.Should().BeEmpty();
+            passwordResult.Violations.Should().BeEmpty();
+            summaryResult.Violations.Should().BeEmpty();

```

Comment:
This doesn't currently test the confirmation page as without having actually submitted this form trying to get to that page just redirects you.
With a function to call somewhere that runs the delete user stored procedure we could maybe test it but that feels outside the scope of this ticket - maybe I should add a todo here to test the confirmation page once we have a was to delete users.

Reply:
I think it's ok as long as we run some accessibility checks on the confirmation page during manual testing 👍